src/PowerShellEditorServices/Services/PowerShell/Execution/SynchronousTask.cs

```diff
@@ -79,16 +80,16 @@ public void ExecuteSynchronously(CancellationToken executorCancellationToken)
                 return;
             }
 
-            using var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(_taskRequesterCancellationToken, executorCancellationToken);
-            if (cancellationSource.IsCancellationRequested)
+            _cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(_taskRequesterCancellationToken, executorCancellationToken);
+            if (executorCancellationToken.IsCancellationRequested)
             {
                 SetCanceled();
                 return;
             }
 
             try
             {
-                TResult result = Run(cancellationSource.Token);
+                TResult result = Run(executorCancellationToken);
```

Comment:
Alright so think of a linked cancellation token source kinda like this:

```csharp
static void MyMethod(CancellationToken callersToken)
{
    CancellationTokenSource myToken = new(millisecondsDelay: 5000);

    // Real method name is "Register", but bare with me
    callersToken.OnCancel(() => myToken.Cancel());
    OtherMethod(myToken.Token);
}
```

That's all linking does. The issue with the WIP here is instead of passing `myToken.Token`, you're passing `callersToken`. So if we call `myToken.Cancel()`, that's not going to do anything. That's also why it looks like it works: it's never cancelled.

I haven't played with it yet, but I would guess we can no longer cancel PSRL like this (e.g. F8/editor commands/continue). If we *can* still cancel PSRL, we should probably look at references for `_taskRequesterCancellationToken` to see if we ever even cancel it.